src/Gui/MainWindow.cpp

```diff
@@ -2079,6 +2081,44 @@ void MainWindow::setPaneText(int i, QString text)
     }
 }
 
+void MainWindow::initialiseSizeLabel()
+{
+    d->sizeLabel = new QToolButton(statusBar());
+    d->sizeLabel->setText(tr("Dimension"));
+    d->sizeLabel->setPopupMode(QToolButton::MenuButtonPopup);
+    QObject::connect(
+        d->sizeLabel, &QToolButton::clicked,
+        d->sizeLabel, &QToolButton::showMenu
+    );
+    d->sizeLabel->setMinimumWidth(120);
+
+    //create the button actions
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Units");
+    int userSchema = hGrp->GetInt("UserSchema", 0);
+    QActionGroup* actionGrp = new QActionGroup(this);
+    actionGrp->setExclusive(true);
+
+    int num = static_cast<int>(Base::UnitSystem::NumUnitSystemTypes);
+    for (int i = 0; i < num; i++) {
+        QAction* action = new QAction(qApp->translate("Gui::Dialog::DlgSettingsUnits", Base::UnitsApi::getDescription(static_cast<Base::UnitSystem>(i))), this);
+        action->setCheckable(true);
+        action->setChecked(i == userSchema);
+        actionGrp->addAction(action);
+
+        QObject::connect(action, &QAction::changed,
+            this, [i, hGrp] {
+                // Set and save the Unit System
+                Base::UnitsApi::setSchema(static_cast<Base::UnitSystem>(i));
+                hGrp->SetInt("UserSchema", i); //Note : saving parameter will trigger EditModeCoinManager::ParameterObserver
```

Comment:
There should be no comment about things that Gui::MainWindow class is not aware.
You can set a general comment such as `Impacted classes have to register an observer` but this is even not needed as this should already be the case.